lua/autorun/server/cfc_ulib_cooldowns.lua

```diff
@@ -1,13 +1,48 @@
 
--- { amount = int, decayRate = int }
--- using a commands adds to its counter if this counter is greater than `amount` you can no longer run the command
+-- { amount = int, decayRate = float }
+-- Using a command adds to its counter. If this counter is greater than `amount`, you can no longer run the command
 -- 1 is subtracted from the counter every `decayRate` seconds
 local ULIB_COOLDOWNS = {
-    ["default"]       = {amount=3, decayRate=2},
-    ["ulx ragdoll"]   = {amount=2, decayRate=5},
-    ["ulx unragdoll"] = {amount=2, decayRate=5},
+    default = {
+        amount = 3,
+        decayRate = 2,
+    },
+    ragdoll = {
+        amount = 2,
+        decayRate = 5,
+    },
+    unragdoll = {
+        amount = 2,
+        decayRate = 5,
+    },
 }
 
+local function setupConVar( name )
+    local cvData = ULIB_COOLDOWNS[name]
+    local amountText = name == "default" and "The max number of times most ULX commands can be used by a player within a timeframe. A value of zero means no limit." or
+        "The max number of times ulx " .. name .. " can be used by a player within a timeframe. A value of zero means no limit."
+    local decayRateText = name == "default" and "The number of seconds it takes for a command's counter to decrease by one." or
+        "The number of seconds it takes for ulx " .. name .. "'s counter to decrease by one."
+
+    local amountCV = CreateConVar( "cfc_ulib_cooldowns_" .. name .. "_amount", cvData.amount, FCVAR_NEVER_AS_STRING, amountText .. " (default " .. cvData.amount .. ")", 0, 50000 )
+    local decayRateCV = CreateConVar( "cfc_ulib_cooldowns_" .. name .. "_decay", cvData.decayRate, FCVAR_NEVER_AS_STRING, decayRateText .. " (default " .. cvData.decayRate .. ")", 0.001, 50000 )
+
+    if name ~= "default" then
+        ULIB_COOLDOWNS[name] = nil
+        name = "ulx " .. name
+        ULIB_COOLDOWNS[name] = {}
+    end
+
+    ULIB_COOLDOWNS[name].amount = amountCV
+    ULIB_COOLDOWNS[name].decayRate = decayRateCV
+end
```

Comment:
Would be fine merging this as-is, but wondered if you had any new ideas for how to structure this?